sklearn/neighbors/_base.py

```diff
@@ -746,9 +746,14 @@ class from an array representing our data set and ask who's
 
         n_samples_fit = self.n_samples_fit_
         if n_neighbors > n_samples_fit:
+            if query_is_train:
+                n_neighbors -= 1
+                inequality_str = "n_neighbors < n_samples"
```

Comment:
There are still [ci failures](https://dev.azure.com/scikit-learn/scikit-learn/_build/results?buildId=42315&view=logs&j=dde5042c-7464-5d47-9507-31bdd2ee0a3a&t=4bd2dad8-62b3-5bf9-08a5-a9880c530c94) Looking at this again, I prefer to keep the `n_samples_fit`, otherwise it may be confusing:

```suggestion
                inequality_str = "n_neighbors < n_samples_fit"
```

I know this is counter to my previous suggestion. I left another comment as a workaround.